swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift

```diff
@@ -154,6 +188,97 @@ public struct SettingsView: View {
     model.load()
     dismiss()
   }
+
+#if os(macOS)
+  func exportLogsButtonTapped() {
+    self.isExportingLogs = true
+
+    let savePanel = NSSavePanel()
+    savePanel.prompt = "Save"
+    savePanel.nameFieldLabel = "Save log zip bundle to:"
+    savePanel.nameFieldStringValue = "firezone-logs.zip"
+
+    guard let window = NSApp.windows.first(where: { $0.identifier?.rawValue.hasPrefix("firezone-settings") ?? false }) else {
+      self.isExportingLogs = false
+      logger.log("Settings window not found. Can't show save panel.")
+      return
+    }
+
+    savePanel.beginSheetModal(for: window) { response in
+      guard response == .OK else {
+        self.isExportingLogs = false
+        return
+      }
+      guard let destinationURL = savePanel.url else {
+        self.isExportingLogs = false
+        return
+      }
+
+      Task {
+        do {
+          try await createLogZipBundle(destinationURL: destinationURL)
+          self.isExportingLogs = false
+          await MainActor.run {
+            window.contentViewController?.presentingViewController?.dismiss(self)
+          }
+        } catch {
+          self.isExportingLogs = false
+          await MainActor.run {
+            // Show alert
+          }
+        }
+      }
+    }
+  }
+#elseif os(iOS)
+  func exportLogsButtonTapped() {
+    self.isExportingLogs = true
+    Task {
+      let logsTempZipFileURL = try await createLogZipBundle()
+      self.isPresentingExportLogShareSheet = true
+    }
+  }
+#endif
+
+  @discardableResult
+  private func createLogZipBundle(destinationURL: URL? = nil) async throws -> URL {
+    let fileManager = FileManager.default
+    guard let logFilesFolderURL = SharedAccess.logFolderURL else {
+      throw SettingsViewError.logFolderIsUnavailable
+    }
+    let zipFileURL = destinationURL ?? fileManager.temporaryDirectory.appendingPathComponent("firezone_logs.zip")
```

Comment:
It might be a good idea to prepend a datetime string to the filename as well. Seconds since epoch could work.

Reply:
Okay. Will do.

Reply:
Done in commit 3bd8d0e. The filename looks something like: `firezone_logs_2023-10-03T152256Z.zip`. This applies for the file exported in the iOS share sheet, and the suggested filename in the macOS save dialog.